### PR TITLE
Shear heating fix

### PR DESCRIPTION
**Pull request by jdannberg**
This improves the approximation of the shear heating in the compressible case. We use the exact solution now instead of the approximation we used before. This especially makes a difference in cases where a Stokes solution is prescribed, but the values for gravity/compressibility/density are not consistent with that solution (because we used the mass conservation in our shear heating formulation). 

I tested this implementation with the test case that is part of this pull request, and found that the new formulation gives us the exact solution of 2.1111 for the shear heating, but the previous implementation resulted in a shear heating of 3. However, in realistic models this difference should be much smaller. 

In addition, this adds a function plugin for prescribing the Stokes solution, which was needed for the test case.

### CODE REVIEW

File: tests/shear-heating-compressible.prm [hunk 1]
```diff
@@ -0,0 +1,118 @@
+# This is a test for the correct behaviour of 
+# the shear heating in the compressible case.
+# The velocity is prescribed using the prescribed 
+# Stokes solution plugin so that the strain rate is 
+# [0 0.5 0.5 2./9.], which means that we can calculate
```
Comment (gassmoeller) [inline tests/shear-heating-compressible.prm]: 2/3?

**Comment by gassmoeller** [comment]
Ok great catch. I think we should add an entry to changes.h since this might change model results. Something like: "Fixed: ... what you wrote above". After that and fixing the other test results this should be good to go. Interesting that mostly the pressure-compatibility tests fail ... I guess they have a relatively large trace of the strain rate tensor.

File: doc/modules/changes.h [hunk 2]
```diff
@@ -6,6 +6,14 @@
  *
  *
  * <ol>
+ *
+ * <li> Changed: We now use the exact formulation with the 
+ * compressible strain rate instead of an approximation 
+ * using the right hand side of the mass conservation equation
+ * to calculate the shear heating. This is more exact in
```
Comment (bangerth) [inline doc/modules/changes.h]: "more exact" is a bit of a funny formulation :-) How about "more accurate"?

**Comment by bangerth** [comment]
Is there a place in the manual that needs to be fixed?

Please change the changes.h entry slightly and merge yourself (no need to re-run the testsuite).

**Comment by jdannberg** [comment]
I do not think that we mention this anywhere in the manual. All the places I have seen only use the equation that I have implemented now, and I guess this is also why nobody knew we even used the approximation...

**Pull request merged**